pandas/core/internals/blocks.py

```diff
@@ -1783,6 +1782,21 @@ def get_values(self, dtype=None):
     def to_dense(self):
         return np.asarray(self.values)
 
+    def to_native_types(self, slicer=None, na_rep="nan", quoting=None, **kwargs):
+        """override to use ExtensionArray astype for the conversion"""
+        try:
+            values = self.values
+            if slicer is not None:
+                values = values[slicer]
+            mask = isna(values)
+            values = values.astype(str)
+            values[mask] = na_rep
+            # we are expected to return a 2-d ndarray
+            return values.reshape(1, len(values))
+        except Exception:
```

Comment:
can you catch something more specific?

Reply:
I know that the Sparse setitem case raises a TypeError. But we don't really know what other EAs might return. Eg in could be possible that some raises an error on `astype(str)`

Reply:
OK.  Should we consider requiring that EASubclass.astype accept str?  Easy to offer default implementation `np.asarray([str(x) for x in self], dtype=object)`